sentinel-adapter/sentinel-apache-dubbo-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/dubbo/BaseSentinelDubboFilter.java

```diff
@@ -38,43 +42,46 @@ public BaseSentinelDubboFilter() {
     static class SentinelDubboListener implements Listener {
 
         public void onResponse(Result appResponse, Invoker<?> invoker, Invocation invocation) {
-            onSuccess(appResponse, invoker);
+            onSuccess(appResponse, invoker, invocation);
         }
 
         //for compatible dubbo 2.7.5 rename onResponse to onMessage
         public void onMessage(Result appResponse, Invoker<?> invoker, Invocation invocation) {
-            onSuccess(appResponse, invoker);
+            onSuccess(appResponse, invoker, invocation);
         }
 
-        private void onSuccess(Result appResponse, Invoker<?> invoker) {
+        private void onSuccess(Result appResponse, Invoker<?> invoker, Invocation invocation) {
             if (DubboConfig.getDubboBizExceptionTraceEnabled()) {
-                traceAndExit(appResponse.getException(), invoker.getUrl());
+                traceAndExit(appResponse.getException(), invoker, invocation);
             } else {
-                traceAndExit(null, invoker.getUrl());
+                traceAndExit(null, invoker, invocation);
             }
         }
 
         @Override
         public void onError(Throwable t, Invoker<?> invoker, Invocation invocation) {
-            traceAndExit(t, invoker.getUrl());
+            traceAndExit(t, invoker, invocation);
         }
 
     }
 
-    static void traceAndExit(Throwable throwable, URL url) {
-        Entry interfaceEntry = (Entry) RpcContext.getContext().get(DubboUtils.DUBBO_INTERFACE_ENTRY_KEY);
-        Entry methodEntry = (Entry) RpcContext.getContext().get(DubboUtils.DUBBO_METHOD_ENTRY_KEY);
-        if (methodEntry != null) {
-            Tracer.traceEntry(throwable, methodEntry);
-            methodEntry.exit();
-            RpcContext.getContext().remove(DubboUtils.DUBBO_METHOD_ENTRY_KEY);
-        }
-        if (interfaceEntry != null) {
-            Tracer.traceEntry(throwable, interfaceEntry);
-            interfaceEntry.exit();
-            RpcContext.getContext().remove(DubboUtils.DUBBO_INTERFACE_ENTRY_KEY);
+    static void traceAndExit(Throwable throwable, Invoker invoker, Invocation invocation) {
+        String methodResourceName = DubboUtils.getResourceName(invoker, invocation, DubboConfig.getDubboConsumerPrefix());
+        Entry[] entries = (Entry[]) RpcContext.getContext().get(methodResourceName);
```

Comment:
As both `SentinelDubboProviderFilter` and `SentinelDubboConsumerFilter` share this method, ` DubboConfig.getDubboConsumerPrefix()` may not correct, when providerPrefix, consumerPrefix have different settings.
If use methodResourceName as key, here we may need to differentiate provider and consumer.

Reply:
emm.. I don't notice this. I will modify it later.